ledger/participant-integration-api/src/main/scala/platform/indexer/parallel/ParallelIndexerFactory.scala

```diff
@@ -61,26 +67,37 @@ object ParallelIndexerFactory {
         "batching-pool",
         Some(metrics.daml.parallelIndexer.batching.executor -> metrics.registry),
       )
-      dbDispatcher <- DbDispatcher
-        .owner(
-          serverRole = ServerRole.Indexer,
-          jdbcUrl = jdbcUrl,
-          connectionPoolSize = ingestionParallelism + 1, // + 1 for the tailing ledger_end updates
-          connectionTimeout = FiniteDuration(
-            250,
-            "millis",
-          ), // 250 millis is the lowest possible value for this Hikari configuration (see HikariConfig JavaDoc)
-          metrics = metrics,
-          connectionAsyncCommitMode = DbType.AsynchronousCommit,
-        )
       toDbDto = backend.UpdateToDbDto(
         participantId = participantId,
         translation = translation,
         compressionStrategy = compressionStrategy,
       )
+      haCoordinator <-
+        if (storageBackend.dbLockSupported && haConfig.enable)
```

Comment:
Well if AFAIK we want to enable HA everywhere we can especially so that it also ensures safety with async commit as well.
Also warn here if the DB is not supporting locking would leave all the H2 participant will start with a warning. I am not sure that is called for.
What do you think @rautenrieth-da @mziolekda @meiersi-da ?

Reply:
Do we want to enable HA everywhere by default? Or automatically choose HA mode depending on the backend?

If I understand correctly, then we can't provide HA if the backend doesn't support advisory locks. IMO we should crash if a participant was started in HA mode and the backend doesn't support HA. And if we choose the HA mode automatically, we should log the chosen HA mode at info level.

Reply:
@rautenrieth-da AFAIK we want to enable this by default. (because it is mainly to provide guards for mutual exclusion, thus not letting multiple indexer start using the same database).

`haConifig.enable` is the feature flag. That is planned to be removed as with the HA epic's cleanup